flutter-gui-tests/testSrc/io/flutter/tests/gui/ProjectCreator.kt

```diff
@@ -0,0 +1,61 @@
+package io.flutter.tests.gui
+
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.testGuiFramework.framework.Timeouts
+import com.intellij.testGuiFramework.impl.*
+import com.intellij.testGuiFramework.utils.TestUtilsClass
+import com.intellij.testGuiFramework.utils.TestUtilsClassCompanion
+import org.fest.swing.exception.WaitTimedOutError
+
+val GuiTestCase.ProjectCreator by ProjectCreator
+
+// Inspired by CommunityProjectCreator
+class ProjectCreator(guiTestCase: GuiTestCase) : TestUtilsClass(guiTestCase) {
+
+  companion object : TestUtilsClassCompanion<ProjectCreator>({ it -> ProjectCreator(it) })
+
+  private val defaultProjectName = "untitled"
+  private val LOG = Logger.getInstance(this.javaClass)
+
+  fun createProject(projectName: String = defaultProjectName, needToOpenMain: Boolean = true) {
+    with(guiTestCase) {
+      welcomeFrame {
+        actionLink("Create New Project").click()
+        GuiTestUtilKt.waitProgressDialogUntilGone(robot = robot(), progressTitle = "Loading Templates",
+                                                  timeoutToAppear = Timeouts.seconds02)
+        dialog("New Project") {
+          jList("Flutter").clickItem("Flutter")
+          button("Next").click()
+          typeText(projectName)
+          button("Finish").click()
+          GuiTestUtilKt.waitProgressDialogUntilGone(robot = robot(), progressTitle = "Creating Flutter Project",
+                                                    timeoutToAppear = Timeouts.seconds03)
+        }
```

Comment:
aside: these wait conditions scare me a little (and bring me back to the kind of test flakes that haunted windowtester)

Reply:
That was my first reaction, too! It isn't that bad, though. The time-out serves as a safety catch. If the dialog never shows we just continue after a couple seconds waiting for it. If it does show, the function waits until it disappears.